File: src/main/java/gr/aueb/cf/e_commerce_app/model/Product.java
package gr.aueb.cf.e_commerce_app.model;

import gr.aueb.cf.e_commerce_app.model.static_data.Category;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "products")
public class Product extends AbstractEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(nullable = false)
    private BigDecimal price;

    private String description;

    @Column(name = "is_active")
    private Boolean isActive;

    private Integer quantity;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

}

File: frontend/src/app/shared/interfaces/product.ts
export interface Product {
  uuid: string;
  name: string;
  description: string;
  price: number;
  quantity: number;
  isActive: boolean;
  category: {
    id: number;
    name: string;
  }
}

export interface ProductInsertDto {
  name: string;
  description: string;
  price: number;
  quantity: number;
  categoryId: number;
}

